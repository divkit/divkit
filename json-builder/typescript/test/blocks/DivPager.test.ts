import {
    divCard,
    DivNeighbourPageSize,
    DivPager,
    DivPageSize,
    DivPercentageSize,
    DivText,
    IDivData,
    fixed,
    reference,
    rewriteRefs,
    templateHelper,
} from '../../src';

describe('DivPager tests', (): void => {
    it('should create simple block', (): void => {
        const block = new DivPager({
            orientation: 'horizontal',
            item_spacing: fixed(10),
            items: [
                new DivText({
                    text: 'Page 1',
                }),
                new DivText({
                    text: 'Page 2',
                }),
            ],
            layout_mode: new DivPageSize({
                page_width: new DivPercentageSize({
                    value: 90,
                }),
            }),
        });

        expect(block).toMatchSnapshot();
    });

    it('should create card with templates', (): void => {
        const templates = {
            div: new DivPager({
                items: [
                    new DivText({
                        text: reference('page1_text'),
                    }),
                    new DivText({
                        text: reference('page2_text'),
                    }),
                ],
                layout_mode: reference('mode'),
            }),
        };

        const divData: IDivData = {
            log_id: 'id',
            states: [
                {
                    state_id: 1,
                    div: templateHelper(templates).div({
                        page1_text: 'Page 1',
                        page2_text: 'Page 2',
                        mode: new DivNeighbourPageSize({
                            neighbour_page_width: fixed(10),
                        }),
                    }),
                },
            ],
        };

        expect(divCard(rewriteRefs(templates), divData)).toMatchSnapshot();
    });
});
