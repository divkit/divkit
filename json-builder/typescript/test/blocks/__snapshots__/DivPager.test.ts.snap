// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DivPager tests should create card with templates 1`] = `
{
  "templates": {
    "div": {
      "type": "pager",
      "items": [
        {
          "type": "text",
          "$text": "page1_text"
        },
        {
          "type": "text",
          "$text": "page2_text"
        }
      ],
      "$layout_mode": "mode"
    }
  },
  "card": {
    "log_id": "id",
    "states": [
      {
        "state_id": 1,
        "div": {
          "type": "div",
          "page1_text": "Page 1",
          "page2_text": "Page 2",
          "mode": {
            "type": "fixed",
            "neighbour_page_width": {
              "type": "fixed",
              "value": 10
            }
          }
        }
      }
    ]
  }
}
`;

exports[`DivPager tests should create simple block 1`] = `
{
  "type": "pager",
  "item_spacing": {
    "type": "fixed",
    "value": 10
  },
  "items": [
    {
      "type": "text",
      "text": "Page 1"
    },
    {
      "type": "text",
      "text": "Page 2"
    }
  ],
  "layout_mode": {
    "type": "percentage",
    "page_width": {
      "type": "percentage",
      "value": 90
    }
  },
  "orientation": "horizontal"
}
`;
